### DISCUSSION TITLE
Vdebug set up

### DISCUSSION THREAD

**Discussion by AaronDovTurkel** [NvChad/NvChad#611] [Q&A] [answered]
Hi! I've been scouring both nvchad and vdebug docs for how to get this to work. But I can't even get the plug-in to load! Has anyone had any success with this?

**Comment by siduck** [comment, marked as answer]
did you install it first?

**Reply by AaronDovTurkel** [reply]
Hi  - packer sync and setting the Python path to use python3 did it for me!

**Comment by siduck** [comment]
if you're looking for a debugger plugin then i would suggest using DAP

**Reply by AaronDovTurkel** [reply]
Im trying to set up DAP now but having issues. Have you done it with PHP? @siduck

**Reply by siduck** [reply]
> Im trying to set up DAP now but having issues. Have you done it with PHP? @siduck

no , issues like?

**Reply by AaronDovTurkel** [reply]
Wasn't sure how to set up the configurations properly.  Ended up doing:
```lua
-- custom/init
   use {
    'mfussenegger/nvim-dap',
  config = function()
    require('custom.plugins.configs.dap')
  end
  }

  use {
    'rcarriga/nvim-dap-ui',
    requires = {"mfussenegger/nvim-dap"},
    config = function() 
      require('dapui').setup()
    end
  }

```

```lua
-- custom/plugins/configs/dap
local dap = require('dap');

dap.configurations.php = {
  {
    type = 'php',
    request = 'launch',
    name = 'Listen for Xdebug',
    port = 9003
  }
}
dap.adapters.php = {    
  type = 'executable',
  command = 'node',
  args = { '/path/to/vscode-php-debug/out/phpDebug.js' } 
}    
  
```

**Reply by AaronDovTurkel** [reply]
Is there a better way to set up the dap configs?
